src/utils.ts

```diff
@@ -114,46 +114,42 @@ export class Utils {
       return nodes.sort((b, a) => (b.x + b.y * column)-(a.x + a.y * column));
   }
 
-  /**
-   * creates a style sheet with style id under given parent
-   * @param id will set the 'gs-style-id' attribute to that id
-   * @param parent to insert the stylesheet as first child,
-   * if none supplied it will be appended to the document head instead.
-   */
-  static createStylesheet(id: string, parent?: HTMLElement): CSSStyleSheet {
-    let style: HTMLStyleElement = document.createElement('style');
-    style.setAttribute('type', 'text/css');
-    style.setAttribute('gs-style-id', id);
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    if ((style as any).styleSheet) { // TODO: only CSSImportRule have that and different beast ??
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (style as any).styleSheet.cssText = '';
-    } else {
-      style.appendChild(document.createTextNode('')); // WebKit hack
-    }
-    if (!parent) {
-      // default to head
-      parent = document.getElementsByTagName('head')[0];
-      parent.appendChild(style);
-    } else {
-      parent.insertBefore(style, parent.firstChild);
+  /** update CSS style on elements with given selector */
+  static updateStyleOnElements(selector: string, styles: {[props: string]: string |  string[] }): void {
+    const elements = Utils.getElements(selector);
+    for (const el  of elements) {
+      Utils.addElementStyle(el, styles)
     }
-    return style.sheet as CSSStyleSheet;
   }
 
-  /** removed the given stylesheet id */
-  static removeStylesheet(id: string): void {
-    let el = document.querySelector('STYLE[gs-style-id=' + id + ']');
-    if (el && el.parentNode) el.remove();
+  /** update CSS style on element */
+  static addElementStyle(el: HTMLElement, styles: {[props: string]: string |  string[] }): void {

```

Comment:
if (!(styles instanceof Object)) return; // not sure this even needed if (!styles) is enough
for (const key in styles) {
  if (!styles.hasOwnProperty(key)) return;
  const style = styles[key];
  if (Array.isArray(style)) { // why do we need this ?
    style.forEach(val => el.style[key] = val);
  } else {
    el.style[key] = style;
  }
}